library/Backends/llvm/src/Juvix/Backends/LLVM/Compilation.hs

```diff
@@ -195,6 +198,14 @@ compilePrimApp ty f xs
 -- | Write LLVM code for a primitive.
 -- TODO: implement other primitives.
 mkPrim ::
+  ( HasThrow "err" Types.Errors m,
+    HasState "blocks" (T Name.Name Types.BlockState) m,
```

Comment:
Do we need all these constraints?

Reply:
Yes, the combination of the new `generateUniqueName` and the `moduleDefinitions` from `globalString` requires all of those.

Reply:
does the `Define` effect entail these, or should we make a more limited effect?

Reply:
Ah, good call -- `Define` does entail them all.  I'll substitute that.